src/runtime/node/process/_process.ts

```diff
@@ -262,3 +266,125 @@ process.hrtime.bigint = function () {
   // Convert milliseconds to nanoseconds
   return BigInt(Date.now() * 1_000_000);
 };
+
+let egid = 0;
+process.getegid = function () {
+  return egid;
+};
+process.setegid = function (val: number) {
+  egid = val;
+};
+
+let euid = 0;
+process.geteuid = function () {
+  return euid;
+};
+process.seteuid = function (val: number) {
+  euid = val;
+};
+
+let gid = 0;
+process.getgid = function () {
+  return gid;
+};
+process.setgid = function (val: number) {
+  gid = val;
+};
+
+let groups = [];
+process.getgroups = function () {
+  return groups;
+};
+process.setgroups = function (val: number[]) {
+  groups = val;
```

Comment:
since this is supported in node only if you have the right privileges I think we should just error with the unimplemented fn as you used elsewhere.

this is what node does unless you run it as root:

```
> process.setgroups([20])
Uncaught Error: EPERM, Operation not permitted
    at process.setgroups (node:internal/bootstrap/switches/does_own_process_state:77:20) {
  errno: 1,
  code: 'EPERM',
  syscall: 'setgroups'
}
```
The same goes for
- [ ] setegid
- [ ] seteuid
- [ ] setgid
- [ ] setuid

Reply:
throwing not implemented for `set` methods makes sense (I have to span a windows machine but I guess that's Node.js behavior too when called on windows/android)

Reply:
Testing on stackblitz, they also throw `EPERM` for `setgroups`. For `get` methods, they return a non zero number which looks safe to me if sensable default to try making codes work as much as possible.